packages/react/src/wallet/infrastructure/SignerFactory.ts

```diff
@@ -92,7 +95,11 @@ export class SignerFactory implements ISignerFactory {
     chainConfig: AddEthereumChainParameter,
   ): PromiseResult<void, WalletConnectionError> {
     try {
-      await signer.provider.send('wallet_switchEthereumChain', [{ chainId: chainConfig.chainId }]);
+      if (signer.provider && signer.provider instanceof JsonRpcProvider) {
```

Comment:
this now modified the logic that relies on the `INCORRECT_CHAIN` error to be thrown.

Previously when switching chain failed we returned the `WalletConnectError` as a failure.
Now when there is no way to switch the network we just move forward making the logic incorrect as we can't move forward connected to the wrong network cause it will cause broadcasting to fail later down the line (chain id is a part of signed message).

So in case we can't try to `switchChain` we should throw the error right away